lib/Cmts.ml

```diff
@@ -280,6 +280,24 @@ let relocate (t : t) ~src ~before ~after =
           r |> Location.Set.remove src |> Location.Set.add after
           |> Location.Set.add before) )
 
+let relocate_match_cmts (t : t) src tok ~whole_loc ~matched_loc =
+  let f map =
+    let f Cmt.{loc= cmt_loc; _} =
+      Source.is_before_match_keyword src tok ~exp_loc:whole_loc ~cmt_loc
```

Comment:
I suggest to break this function. `tok` and `whole_loc` are constant, which mean that the `tokens_at` call could be made only once outside of the loop.
The function `f` would then only need to do the comparison, which is as explicit and more efficient.

Reply:
Good point, I modified the api of source.mli and simplified the relocating function